### ISSUE TITLE
No English localization with 24h date time format (only am/pm)  

### ISSUE THREAD

**Issue by marcusm7** [opencast/opencast-admin-interface#922]
Opencast 14 allowed to use en-gb to have an english ui and 24h time format. This may have been a bug in OC before, but it was also a very nice feature! A time format which starts counting with 12 o'clock isn't ideal for everyone. ;)   
  
Can we have an 24h date format + english translation, please?

**Comment by Arnei** [comment]
#1509 brings back 24 hour cycle to en-GB as a side effect, because it bases the formatting on the locale. Maybe that is good enough for this issue?

We could also add an extra ui element where users can set their preferred datetime format independent from the locale, but that seems very overkill to me.

**Comment by marcusm7** [comment]
> [#1509](https://github.com/opencast/opencast-admin-interface/pull/1509) brings back 24 hour cycle to en-GB as a side effect, because it bases the formatting on the locale. Maybe that is good enough for this issue?

En-GB brings back 24h time format + a more sane dd/mm/yyyy date format. That is nice. :)  
  
Is it hard to add a user preferences page for time format, dark/ light mode, pagination-setting, preferred language and so on? I can't oversee the side effects and development and maintenance effort. What's the estimate?  
  
I just like to select date format (i.e. YYYY-MM-dd hh24:mm:ss), language and so on, based on the application (Opencast, Youtube, Nextcloud).

**Comment by mtneug** [comment]
> Is it hard to add a user preferences page for time format, dark/ light mode, pagination-setting, preferred language and so on? I can't oversee the side effects and development and maintenance effort. What's the estimate?

Why not let the OS / browser worry about the preference pages and simply rely on the values from there. Having our own would allow users to set custom values just for Opencast, but who needs that?

**Comment by marcusm7** [comment]
TLDR:  
I need that. But I'm not desperate. :)  
In OC I can select the language, which affects the date-time-format. I wasn't able to find a language setting for english + YYYY-MM-dd + 24h time format. That is all what I'm asking for - others might need a different combination.  
  
The issue is not solved from my point of view. But I understand, if you say it is OK the way it is, because of other constraints. 

Arguments:  
- Call me old fashioned, but the date-time-format is application or even use case specific. Sometimes I need YYYY-MM-dd HH24:mm:ss, sometimes dd.MM.YY HH24:mm - it depends on the use case and the user.
- A Locale is a bundle of settings: language, date-time-format, font-settings and so on...). The application can assume why I selected a certain locale, but it can't be sure. And at this point I would like to have the option to configure the application to my needs. Youtube shouldn't translate german videos just because my browser is set to en_gb. Collabora shouldn't use dark mode for documents, just because the browser says so.  
- OC allows to change the language, which is great.  But defacto it is the locale. But why not allow a choice about the date-time-format? Or allow to add custom locales in the language dropdown. ;) 
  
To be clear. I can deal with the current state. My discussion is about good enough or good (from my point of view). The discussion about this is very subjective and influenced by personal preferences. I can give feedback from my perspective. I understand that everyone has their own perspective. And there are more constraints such as development and maintenance costs, task prioritization and even training costs for users are a factor.

**Comment by mtneug** [comment]
> Call me old fashioned, but the date-time-format is application or even use case specific. Sometimes I need YYYY-MM-dd HH24:mm:ss, sometimes dd.MM.YY HH24:mm - it depends on the use case and the user.

I never imagined that people would want to have different date formats in different applications. It would be interesting to hear reasons for this.

I totally see your point regarding languages in YouTube and automatic dark mode. That has frustrated me at times as well. So, I'm not opposed to adding configuration options in Opencast for data formats, but just though it would be easier to follow the browser local settings. It seems I was mistaken.

**Comment by marcusm7** [comment]
> I never imagined that people would want to have different date formats in different applications. It would be interesting to hear reasons for this.

I have a personal preference for ISO8601 YYYY-MM-ss hh24:mm:ss because legibility and natural sort order - sv_SE.UTF-8 would match this. When I write a letter, I have to follow local conventions (dd.MM.YYYY, de_DE.UTF-8). Two use cases with different requirements.  
  
Edit: As a matter of fact sv_SE works as expected in https://github.com/opencast/opencast-admin-interface/pull/1509. As long as no one ever provides a swedish localization, I can use this. But I wouldn't set it as Browser Locale because of other domains.

**Comment by Arnei** [comment]
I believe #1564 (formerly #1509) has fixed the main issue here, so I would like to close this.

Other issues and feature ideas that emerged during the discussion are probably better placed in their own issue tickets, as they are likely to get lost among the text otherwise.
